rfcs/000-consolidate-amundsen-microservices.md

```diff
@@ -0,0 +1,154 @@
+- Feature Name: consolidate_amundsen_microservices
+- Start Date: 2021-03-23
+- RFC PR: [amundsen-io/rfcs#30](https://github.com/amundsen-io/rfcs/pull/30)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Consolidate Amundsen Microservices
+
+## Summary
+
+The idea is to (eventually) deprecate the split of `amundsenmetadatalibrary`, `amundsenfrontendlibrary`, `amundsensearchlibrary` and `amundsencommon` repositories, and merge them all into one mono repository. The split will instead be based on the backend and frontend.
+
+Following will be the repositories (and packages) of Amundsen after the change.
+- amundsen (frontend, metadata, search, and common repositories)
+- amundsendatabuilder (Same as today's databuilder repository)
+
+There will not be a separate `amundsenmetadatalibrary`, `amundsenfrontendlibrary`, `amundsensearchlibrary` or `amundsencommon` packages.
```

Comment:
I think another reviewer has mentioned that maybe this goal should be done as a second stage? The thought here being that we can still get the benefits to productivity from the mono-repo but not take on all of the migration challenges quite yet.